src/commands/app/upgrade.tsx

```diff
@@ -0,0 +1,230 @@
+            this.flags["target-version"],
+        );
+
+      if (targetVersionMatchFromCandidates) {
+        targetAppVersion = targetVersionMatchFromCandidates;
+      } else {
+        process.addInfo(
+          <Text>
+            The given target upgrade version does not seem to be a valid upgrade
+            candidate.
+          </Text>,
+        );
+        targetAppVersion = await forceTargetVersionSelection(
+          process,
+          this.apiClient,
+          targetAppVersionCandidates,
+          currentApp,
+          currentAppVersion,
+        );
+      }
+    } else {
+      targetAppVersion = await forceTargetVersionSelection(
+        process,
+        this.apiClient,
+        targetAppVersionCandidates,
+        currentApp,
+        currentAppVersion,
+      );
+    }
+
+    if (!targetAppVersion) {
+      process.error("Target app version could not be determined properly.");
+      return;
+    }
+
+    if (!this.flags.force) {
+      const confirmed: boolean = await process.addConfirmation(
+        <Text>
+          Confirm upgrading {currentApp.name}{" "}
+          {currentAppVersion.externalVersion} (
+          {currentAppInstallation.description}) to version{" "}
+          {targetAppVersion.externalVersion}
+        </Text>,
+      );
+      if (!confirmed) {
+        process.addInfo(<Text>Upgrade will not be triggered.</Text>);
+        process.complete(<></>);
+        ux.exit(1);
+      }
+    } else {
+      process.addInfo(
+        <Text>
+          Commencing upgrade of {currentApp.name}{" "}
+          {currentAppVersion.externalVersion} (
+          {currentAppInstallation.description}) to Version{" "}
+          {targetAppVersion.externalVersion}.
+        </Text>,
+      );
+    }
+
+    const patchAppTriggerResponse =
+      await this.apiClient.app.patchAppinstallation({
+        appInstallationId,
+        data: { appVersionId: targetAppVersion.id },
+      });
+
+    assertStatus(patchAppTriggerResponse, 204);
+
+    let successText: string;
+
+    if (this.flags.wait) {
+      await waitUntilAppIsUpgraded(this.apiClient, process, appInstallationId);
+      successText =
+        "The upgrade finished successfully. Please check if everything is in its place. 🔎";
+    } else {
+      successText = "The upgrade has been started. Buckle up! 🚀";
+    }
+    await process.complete(<Success>{successText}</Success>);
+  }
+
+  protected render(): ReactNode {
+    return true;
+  }
+}
+
+async function forceTargetVersionSelection(
+  process: ProcessRenderer,
+  apiClient: MittwaldAPIV2Client,
+  targetAppVersionCandidates: AppAppVersion[],
+  currentApp: AppApp,
+  currentAppVersion: AppAppVersion,
+): Promise<AppAppVersion | undefined> {
+  const targetAppVersionString = await process.addSelect(
+    `Please select target upgrade for your ${currentApp.name} ${currentAppVersion.externalVersion} from one of the following`,
+    [
+      {
+        value: "latest",
+        label: "latest",
+      },
```

Comment:
Do we actually need this special "latest" option? The actual latest version should be the first in the candidate list anyway, shouldn't it?

Reply:
You aren't wrong - I just thought that giving the latest option would be more in unison with the option to declare the `--target-version` as `latest`, this might be personal preference though.

Also, the list returned by `apiClient.app.listUpdateCandidatesForAppversion()` is not ordered by semver.

- Ordering it during runtime for the interactive selection is not a big task but I'd argue its the wrong layer to do it on.
- Ordering it for the `list-upgrade-candidates`command might be though since this is a direct digest of the api-client-response. 
- Ordering it in the api-client before returning it might be a weird way to handle it since the api-client should contain as little logical digestion as possible.

I would argue this should already be returned by the api as a semver sorted list.
And as soon as the list os actually ordered, the `latest` entry might as well be purged.

Reply:
sort implemented